CIP-0106/README.md

```diff
@@ -0,0 +1,173 @@
+
+Specifically, apps building transactions need to be able to get the following information from the wallet:
+- Script descriptor 
+- Script Requirements list 
+- Collateral donator (since native script based addresses cannot provide collateral for transactions)
+
+Additionally, apps need to be able to submit a transaction to the wallet for signing in an asynchronous manner, as gathering of signatures can take a long time and each wallet provider will have its own way of handling this process. 
+
+Finally, the signTx() and signData() endpoints will have to be disabled when using this extension since they are not compatible with native script based addresses.
+
+### Rationale for the required data
+
+- Script descriptor:
+	Any transaction consuming a UTxO from a native script based address must attach the corresponding script. 
+- Script Requirements list:
+	-- dApps need to know the number of signers to calculate the fees correctly.
+	-- Including the correct "required signers" brings UX improvements.
+	-- dApps need to know if the script has any before-after requirements to set the correct validBefore and TTL values.
+- Collateral donator
+
+Native script based addresses cannot provide collateral for Plutus transactions 
+
+## Specification
+
+### Data Types
+
+#### KeyHash
+
+A hex-encoded string of the corresponding bytes. This represents the hash of the public key used to sign transactions.
+
+```ts
+type KeyHash = String
+```
+
+#### ScriptRequirements
+
+```ts
+type ScriptRequirementsCode = {
+    Signer: 1,
+    Before: 2,
+    After: 3,
+}
+type ScriptRequirement = {
+    code: ScriptRequirementsCode,
+    value: KeyHash|number,
+}
+```
+
+### Aditional Error Types
+
+#### CompletedTxError
+
+```ts
+CompletedTxErrorCode = {
+	NotFound: 1,
+	NotReady: 2
+}
+```
+
+* NotFound - The transaction with the given id was not found.
+* NotReady - The transaction with the given id is not ready yet. 
+
+## Motivation: why is this CIP necessary?
+In order to facilitate future dApp development, we will need a way for dApps to communicate with multisig wallets. Given the unique complexities of native script based addresses, special provisions need to be made to make the connector compatible with them.
+
+Specifically, apps building transactions need to be able to get the following information from the wallet:
+
+- Script descriptor
+- Script Requirements list
+- Collateral donator (since native script based addresses cannot provide collateral for transactions)
+
+Additionally, apps need to be able to submit a transaction to the wallet for signing in an asynchronous manner, as gathering of signatures can take a long time and each wallet provider will have its own way of handling this process.
+
+Finally, the signTx() and signData() endpoints will have to be disabled when using this extension since they are not compatible with native script based addresses.
+
+## Rationale for the required data
```

Comment:
```suggestion
### Rationale for the required data
```

Reply:
A continuation of https://github.com/cardano-foundation/CIPs/pull/617#discussion_r1600416733.